src/server/internalServer.cpp

```diff
@@ -778,7 +778,7 @@ std::unique_ptr<Response> InternalServer::handle_content(const RequestContext& r
   } catch (const std::out_of_range& e) {}
 
   if (archive == nullptr) {
-    std::string searchURL = m_root+"/search?pattern="+pattern; // Make a full search on the entire library.
+    std::string searchURL = kiwix::urlEncode(m_root + "/search?pattern=" + pattern); // Make a full search on the entire library.
```

Comment:
There is another flaw here, it is possible to forge a specific url with:
`tld/foo&content=bar`, we will generate a link to `tld/search?pattern=foo&content=bar`.
It is less important that inserting a js in the page but we should fix it anyway.ç
The solution is probably to urlEncode only the pattern with `encodeReserved` at true.